requests_gssapi/gssapi_.py

```diff
@@ -97,6 +97,9 @@ def __init__(self, mutual_authentication=REQUIRED, service="HTTP",
 
         if self.creds and self.principal:
             raise RuntimeError("creds and principal are mutually exclusive")
+        if type(service) == gssapi.names.Name and hostname_override != None:
```

Comment:
should we retain hostname_override at all ?
shouldn't we just deal it in the compat wrapper for the old class to translate it into a gssapi service name before calling into the HTTPSPNEGOAuth initializer ?

Reply:
If we want to make it easy to pass in a string instead of a gssapi name type then we could change the service string to allow to pass in "HTTP@hostname", it sounds silly to have a "hostname_pverride as a separate argument when you can easily use the "service", which I would perhaps rename to "target", but that may be going too far.